src/reports/components/report-sections/make-details-section-fc.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { css } from '@uifabric/utilities';
+import { CommentIcon } from 'common/icons/comment-icon';
+import { DateIcon } from 'common/icons/date-icon';
+import { UrlIcon } from 'common/icons/url-icon';
+import { NamedFC, ReactFCWithDisplayName } from 'common/react/named-fc';
+import { TargetAppData } from 'common/types/store-data/unified-data-interface';
+import * as React from 'react';
+import { SectionProps } from './report-section-factory';
+
+export type DetailsSectionProps = Pick<
+    SectionProps,
+    'targetAppInfo' | 'deviceName' | 'description' | 'scanDate' | 'toUtcString'
+>;
+
+export type ScanDetailInfo = {
+    label: string;
+    content: string | JSX.Element;
+};
+
+export function makeDetailsSectionFC(
+    getUrlOrDeviceItemInfo: (appInfo: TargetAppData, device: string) => ScanDetailInfo,
+): ReactFCWithDisplayName<DetailsSectionProps> {
+    return NamedFC<DetailsSectionProps>('DetailsSection', props => {
+        const { targetAppInfo, deviceName, description, scanDate, toUtcString } = props;
+
+        const createListItem = (
+            icon: JSX.Element,
+            label: string,
+            content: string | JSX.Element,
+            contentClassName?: string,
+        ) => (
+            <li>
+                <span className="icon" aria-hidden="true">
+                    {icon}
+                </span>
+                <span className="screen-reader-only">{label}</span>
+                <span className={css('text', contentClassName)}>{content}</span>
+            </li>
+        );
+
+        const scanDateUTC: string = toUtcString(scanDate);
+        const showCommentRow = !!description && description !== '';
+        const urlOrDeviceItemInfo: ScanDetailInfo = getUrlOrDeviceItemInfo(
```

Comment:
The general idea here is great; it lets us specify the content of this list item in separate components that can target either unified or web.

However, there is a bit of an issue in this class particularly: this class is generic and will be utilized in both scenarios. The device name however is only for the unified scenario, so this is making an assumption of what data we have and the reason this works fine _now_ is JS will be comfortable with null properties. However, someone coming back to this code later might not understand that deviceName might sometimes be null and that this is not a good place to touch it in any significant way.

Ideally, in this case, we shouldn't touch device name because this generic component doesn't _really_ have to know that exists or not, right? All it's doing is passing it to the `getUrlOrDeviceItemInfo` method. To resolve this, we simply don't spread the scanMetadata at this level, since that object contains properties that might not be available in all scenarios for this **generic** component.

That means modifying the `SectionProps` such that it contains the 'scanMetadata` property rather than being combined with the `ScanMetaData` type.

Reply:
I'd also suggest re-naming the `getUrlOrDeviceItemInfo` method. As far as this class is concerned it doesn't matter what is returned by that method as long as it has the expected return types. If you want to be more specific, ideally we'd capture broadly what is happening here. Maybe something like: `getDisplayedScanTargetInfo` or something?